housekeeper/store/api/find.py

```diff
@@ -76,12 +77,25 @@ def version(
             bundle_name=bundle,
         ).first()
 
-    def tag(self, name: str) -> Tag:
-        """Fetch a tag from the database."""
-        return self.Tag.filter_by(name=name).first()
+    def tag(self, tag_name: str = None) -> Tag:
```

Comment:
Naming - the function should have a more descriptive name. Like get_tag_by_name or something similar

Reply:
To discuss tomorrow in meeting. If `tag` has to be changed, then also `file`, `bundle` and `version`.

Reply:
This case differs since only one argument is provided.

The other ones (file, bundle, version) need to be refactored as well since the if-statements make the code harder to read, test and easier to break.

Reply:
Yes this had to be changed to a more explicit get for all of them